pkg/cmd/roachtest/tests/admission_control_intent_resolution.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package tests
+
+import (
+	"context"
+	gosql "database/sql"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/grafana"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/spec"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/install"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/prometheus"
+	"github.com/stretchr/testify/require"
+)
+
+// This test sets up a 1 node CRDB cluster on an 8vCPU machine, runs a txn
+// that creates a huge number of intents, and then commits to resolve those
+// intents. When intent resolution is not subject to admission control, the
+// LSM gets overloaded and has > 50 sub-levels.
+func registerIntentResolutionOverload(r registry.Registry) {
+	r.Add(registry.TestSpec{
+		Name:      "admission-control/intent-resolution",
+		Owner:     registry.OwnerAdmissionControl,
+		Timeout:   time.Hour,
+		Benchmark: true,
+		Tags:      registry.Tags(`weekly`),
+		// Second node is solely for Prometheus.
```

Comment:
We don't need to BTW - even running prometheus on the same node, or not at all, is ok. These runs happen on GCE and we now have that global prometheus instance that knows how to scrape every endpoint.